contracts/Options/Marginable.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.17;
+
+import "../StableCoins/ERC20StableCoin.sol";
+import "../DataFeeder/ReferenceLevel.sol";
+
+abstract contract Marginable {
+    address internal _broker;
+    uint256 internal _marginAmount;
+    uint16 internal _initialMargin; // Initial margin in percentage
+    uint16 internal _maintenanceMargin; // Maintenance margin in percentage
+    ERC20StableCoin internal _marginToken;
+    ERC20StableCoin internal _contractToken;
+    RefernceLevel internal _fxRate;
+
+    constructor(
+        address broker,
+        ERC20StableCoin marginToken,
+        ERC20StableCoin contractToken,
+        RefernceLevel fxRate,
+        uint16 initialMargin,
+        uint16 maintenanceMargin
+    ) {
+        _broker = broker;
+        _initialMargin = initialMargin;
+        _maintenanceMargin = maintenanceMargin;
+        _marginToken = marginToken;
+        _contractToken = contractToken;
+        _fxRate = fxRate;
+        _marginAmount = 0; // Initialize margin as 0 before actual margin calls
+    }
+
+    /**
+     * @dev Throws if called by any account other than the broker.
+     */
+    modifier onlyBroker() {
+        require(_broker == msg.sender, "Marginable: caller is not the broker");
+        _;
+    }
+
+    event MarginAdjusted(uint256 amount, uint256 currentAmount);
+
+    function calInitialMargin(
```

Comment:
'Cal' = 'Calc' or 'Calculate' ? - long method names are better - code is more readble